### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by yahiakr**
## Update dependencies

Checklist: #576.

* Update `Mongoose` to `v6.0.13`:
    * Fix: change population function in `postSave` hook.
    * Fix the failed test `map a geo_point field of an nested referenced schema as a geo_point` in `mapping.test.ts` due to the package update.
* Update `@elastic/elasticsearch` to `v7.15.0`:
    * Fix: update the renamed types/interfaces.
* Update `lodash` to `v4.17.21`.

### CODE REVIEW

**Comment by yahiakr** [comment]
About `Jest`, the new `27.x` version does not support done callback and async test simultaneously (see this: [#10529](https://github.com/facebook/jest/issues/10529)). 

As we are using both of them in event-based tests (e.g: to test `synchronize` method), and since Jest is a dev dependency, i think we can keep the current version for now, and focus on updating it after realising the 5.0 version.

File: package.json [hunk 1]
```diff
@@ -17,21 +17,19 @@
   "main": "dist/index.js",
   "types": "index.d.ts",
   "dependencies": {
-    "@elastic/elasticsearch": "7.13.0",
-    "lodash": "4.17.21",
-    "mongoose": "5.13.13"
+    "@elastic/elasticsearch": "^7.15.0",
+    "lodash": "^4.17.21",
+    "mongoose": "^6.0.13"
```
Comment (guumaster) [inline package.json:22]: please pin dependencies to a fixed version (without ^ prefix).

File: test/alternative-index-method.test.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { Tweet } from './models/tweet'
 describe('Index Method', function () {
 
   beforeAll(async function () {
-    await mongoose.connect(config.mongoUrl, config.mongoOpts)
+    await mongoose.connect(config.mongoUrl)
```
Comment (guumaster) [inline test/alternative-index-method.test.ts:8]: `mongoose.connect()` doesn't accept options anymore?

File: test/filtering.test.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ describe('Filter mode', function () {
 
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore: callback type
-  it('should not index action genre', async function (done) {
+  it('should not index action genre', async function () {
```
Comment (guumaster) [inline test/filtering.test.ts:67]: for this new code you can remove both comment lines preceding it

File: test/index.test.ts [hunk 4]
```diff
@@ -360,7 +360,7 @@ describe('indexing', function () {
 
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore: callback type
-    beforeAll(async function (done) {
+    beforeAll(async function () {
```
Comment (guumaster) [inline test/index.test.ts:363]: same, remove two comment lines above

File: test/index.test.ts [hunk 5]
```diff
@@ -578,7 +581,7 @@ describe('indexing', function () {
 
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore: callback type
-    it('should save but not index', async function (done) {
+    it('should save but not index', async function () {
```
Comment (guumaster) [inline test/index.test.ts:584]: remove two comment lines above

**Review by guumaster — CHANGES_REQUESTED** [review]

File: test/alternative-index-method.test.ts (hunk 2 above)
Comment (yahiakr) [reply test/alternative-index-method.test.ts:8]: Yes, `mongoose.connect()` does not accept the passed options anymore.

**Review by guumaster — APPROVED** [review]

**Pull request merged**